### PR TITLE
Support `_autoupdate` url param for chart preview

### PR DESCRIPTION
**Pull request by mournfulCoroner**

### CODE REVIEW

File: src/ui/components/Widgets/Chart/hooks/useLoadingChart.ts [hunk 1]
```diff
@@ -496,6 +508,34 @@ export const useLoadingChart = (props: LoadingChartHookProps) => {
...
+
+            reloadTimeout = setTimeout(() => reloadChart(), autoupdateIntervalMs);
+        }
+    }, [loadChartData, autoupdateInterval]);
+
+    // need update timeout if it's first render with autoupdateInterval = 0, or if visibility of page is changed
+    React.useEffect(() => {
+        clearTimeout(reloadTimeout);
+        reloadChart();
+    }, [isPageHidden, autoupdateInterval]);
```
Comment (Marginy605) [inline src/ui/components/Widgets/Chart/hooks/useLoadingChart.ts:537]: I suppose you need `reloadChart` in deps

File: src/ui/components/Widgets/Chart/hooks/useLoadingChart.ts [hunk 2]
```diff
@@ -496,6 +508,34 @@ export const useLoadingChart = (props: LoadingChartHookProps) => {
...
+                reloadIntervalRemains <= 0 &&
+                reloadTimeout
+            ) {
+                lastReloadAt = new Date().getTime();
+                loadChartData();
+            }
+
+            reloadTimeout = setTimeout(() => reloadChart(), autoupdateIntervalMs);
+        }
+    }, [loadChartData, autoupdateInterval]);
```
Comment (Marginy605) [inline src/ui/components/Widgets/Chart/hooks/useLoadingChart.ts:531]: I suppose you need to add `isPageHiddenCurrent` in deps, perhaps `reloadChart` too.

File: src/ui/components/Widgets/Chart/hooks/useLoadingChart.ts [hunk 3]
```diff
@@ -79,12 +79,19 @@ export type LoadingChartHookProps = {
...
+    isPageHidden?: boolean;
+    autoupdateInterval?: number;
 };
 
 const WIDGET_LOADING_VISIBLE_OFFSET = 300; // '300px'; // offset before div for start loading TODO - return here CHARTS-7043
 const CHART_DEBOUNCE_TIMEOUT = 100;
 const EXCLUDE_CHART_WITH_AXIS_FOR_MENU_RERENDER = ['map'];
 
+// need to track the current values in setTimeout callback of reloadChart
+let isPageHiddenCurrent: undefined | boolean = false;
```
Comment (Marginy605) [inline src/ui/components/Widgets/Chart/hooks/useLoadingChart.ts:91]: Let's use ref instead

**Review by Marginy605 — CHANGES_REQUESTED** [review]

File: src/ui/components/Widgets/Chart/hooks/useLoadingChart.ts (hunk 1 above)
Comment (mournfulCoroner) [reply src/ui/components/Widgets/Chart/hooks/useLoadingChart.ts:551]: After the experiments, we decided to leave it as it is

File: src/ui/components/Widgets/Chart/hooks/useLoadingChart.ts (hunk 2 above)
Comment (mournfulCoroner) [reply src/ui/components/Widgets/Chart/hooks/useLoadingChart.ts:542]: Now it's part of the ref

**Review by Marginy605 — APPROVED** [review]

**Pull request merged**